playbooks/secrets-init.yml

```diff
@@ -52,13 +52,9 @@
             gs_mode: "plugin"
           when: vault_init | default('') == 'plugin'
 
-        - name: Populate secrets.yml with missing secrets
-          ansible.builtin.shell: "{{ base_folder }}/scripts/generate-secret.sh -s {{ item }} -m {{ gs_mode }} >> {{ secrets_file }}"
-          changed_when: true
-          when: not lookup('vars', item, default=False)
-          tags:
-            - molecule-idempotence-notest
-          loop:
+        - name: Generate loop of mandatory passwords when empty inventory is provided
+          ansible.builtin.set_fact:
+            password_loop:
             - ca_signing_key_passphrase
             - certs_p12_passphrase
```

Comment:
I'm not sure in which case would we get an empty inventory? AFAIR our playbook is (unfortunately) very much tied to an inventory with the required structure of groups

Reply:
Secrets-init is used without inventory in pipeline test scenarios:
https://github.com/Alfresco/alfresco-ansible-deployment/blob/master/tests/molecule_it/script.sh#L8